src/daemon/plugin.c

```diff
@@ -2234,8 +2239,35 @@ void plugin_log(int level, const char *format, ...) {
 
     le = le->next;
   }
+} /* void _daemon_log */
+
+void plugin_log(int level, const char *format, ...) {
+  char msg[1024] = "";
+
+  char const *name = plugin_get_ctx().name;
+  if (name != NULL)
+    snprintf(msg, sizeof(msg), "%s plugin: ", name);
+
+  va_list ap;
+  va_start(ap, format);
+  vsnprintf(msg + strlen(msg), sizeof(msg) - strlen(msg), format, ap);
+  msg[sizeof(msg) - 1] = 0;
+  va_end(ap);
+
+  _daemon_log(level, msg);
 } /* void plugin_log */
 
+void daemon_log(int level, const char *format, ...) {
+  char msg[1024] = "";
```

Comment:
just a thought, but why limit the length to 1024 bytes?

Reply:
>"640k ought to be enough for anybody" 

I don't know... Historical reasons..
)

Reply:
would be nice to allocate it dynamically using alloca or vla's, stacks are large enough these days.